pkg/storage/tsdb/bucket_client.go

```diff
@@ -32,6 +32,11 @@ func NewBucketClient(ctx context.Context, cfg BucketConfig, name string, logger
 		return nil, err
 	}
 
+	// Wrap the client with any provided middleware
+	for _, wrap := range cfg.Middlewares {
+		client = wrap(client)
+	}
+
```

Comment:
I would put this *after* wrapping into tracing bucket, so that spans only cover "real" bucket usage, not wrappers.

Reply:
👍